### DISCUSSION TITLE
huge repository for a library

### DISCUSSION THREAD

**Discussion by soloturn** [bulletphysics/bullet3#3694] [General]
tried to build arch aur package: https://aur.archlinux.org/packages/bullet3-git/

the clone took forever as the git repository is nearly 300MB, which was was surprising, for such a beatiful piece of code. to find large objects i used:
```
git rev-list --objects --all \
| git cat-file --batch-check='%(objecttype) %(objectname) %(objectsize) %(rest)' \
| sed -n 's/^blob //p' \
| sort --numeric-sort --key=2 \
| cut -c 1-12,41- \
| $(command -v gnumfmt || echo numfmt) --field=2 --to=iec-i --suffix=B --padding=7 --round=nearest
```

**Comment by diegoferigo** [comment]
There's no way to amend the history of a git repo without breaking all the existing clones and forks. In this case, we have to live with this big size. For packaging purpose, you can clone the repo with `--depth 1`.
